### DISCUSSION TITLE
Nvidia log spam

### DISCUSSION THREAD

**Discussion by mkpaz** [wezterm/wezterm#3216] [General]
Probably not a bug, but I faced Nvidia issue I'd like to solve, but don't know how. Every time I launch `journalctl` I see the bunch of errors.

```text
Mar 07 20:51:13 hydra /usr/libexec/gdm-x-session[13154]: (--) NVIDIA(GPU-0):
Mar 07 20:51:13 hydra /usr/libexec/gdm-x-session[13154]: (--) NVIDIA(GPU-0): DFP-3: 165.0 MHz maximum pixel clock
Mar 07 20:51:13 hydra /usr/libexec/gdm-x-session[13154]: (--) NVIDIA(GPU-0): DFP-3: Internal TMDS
Mar 07 20:51:13 hydra /usr/libexec/gdm-x-session[13154]: (--) NVIDIA(GPU-0): DFP-3: disconnected
Mar 07 20:51:13 hydra /usr/libexec/gdm-x-session[13154]: (--) NVIDIA(GPU-0):
Mar 07 20:51:13 hydra /usr/libexec/gdm-x-session[13154]: (--) NVIDIA(GPU-0): DFP-2: 1440.0 MHz maximum pixel clock
Mar 07 20:51:13 hydra /usr/libexec/gdm-x-session[13154]: (--) NVIDIA(GPU-0): DFP-2: Internal DisplayPort
Mar 07 20:51:13 hydra /usr/libexec/gdm-x-session[13154]: (--) NVIDIA(GPU-0): DFP-2: disconnected
Mar 07 20:51:13 hydra /usr/libexec/gdm-x-session[13154]: (--) NVIDIA(GPU-0):
Mar 07 20:51:13 hydra /usr/libexec/gdm-x-session[13154]: (--) NVIDIA(GPU-0): DFP-1: 165.0 MHz maximum pixel clock
Mar 07 20:51:13 hydra /usr/libexec/gdm-x-session[13154]: (--) NVIDIA(GPU-0): DFP-1: Internal TMDS
Mar 07 20:51:13 hydra /usr/libexec/gdm-x-session[13154]: (--) NVIDIA(GPU-0): DFP-1: disconnected
Mar 07 20:51:13 hydra /usr/libexec/gdm-x-session[13154]: (--) NVIDIA(GPU-0):
Mar 07 20:51:13 hydra /usr/libexec/gdm-x-session[13154]: (--) NVIDIA(GPU-0): BenQ GW2270 (DFP-0): 330.0 MHz maximum pix>
Mar 07 20:51:13 hydra /usr/libexec/gdm-x-session[13154]: (--) NVIDIA(GPU-0): BenQ GW2270 (DFP-0): Internal TMDS
Mar 07 20:51:13 hydra /usr/libexec/gdm-x-session[13154]: (--) NVIDIA(GPU-0): BenQ GW2270 (DFP-0): connected
```

I googled a bit I found out it's kinda expected behavior and Nvidia [sends this garbage](https://forums.developer.nvidia.com/t/systemd-is-logging-list-of-nvidia-gpu-interfaces-each-30-seconds-all-the-time/66545/4) to the log every time something pings X11 info via `xrandr`. I thought it's some of the Gnome extensions, but surprisingly it's Wezterm itself. So, the sequence is like this:
- I want to check logs
- I launch Wezterm
- Wezterm pings `xrandr`
- My log output always starts with several lines of garbage that looks like some kind of error.

![image](https://user-images.githubusercontent.com/25181998/223494894-cc48a833-d7b3-4acc-8946-373e3d00dd94.png)

So, I'd like to ask for maybe some kind of workaround? I only use a single monitor.

**Comment by wez** [comment]
wezterm needs to query this information on startup, and may query it when you use the https://wezfurlong.org/wezterm/config/lua/wezterm.gui/screens.html function.

This cannot be disabled, sorry!

Workarounds:

* Use wayland (but I don't recommend it: nvidia and wayland are still not a great combination)
* Use the nouveau drivers (again: not a great recommendation)
* Use a different GPU vendor (this was my solution to a variety of nvidia+linux issues)

**Comment by mkpaz** [comment]
Thanks for the answer. I thought that maybe changing systemd log level would help, but thanks to the Nvidia engineers it's not.
